mix.exs

```diff
@@ -27,7 +27,7 @@ defmodule UeberauthTwitter.Mixfile do
     [
      {:httpoison, "~> 0.7"},
      {:oauther, "~> 1.1"},
-     {:poison, "~> 1.3 or ~> 2.0"},
+     {:poison, "~> 1.3 or ~> 2.0 or ~> 3.0 or ~> 4.0"},
```

Comment:
Well, actually I have no best decision.

Problem with `:override` can appear, when you use in `deps` two projects, that rely on `poison`, but don't use `poison` itself. This should be explicitly pointed in README.

For the other hand, may be the best solution is to use only last version of poison, and let users to deal with legacy themselves.

Or, may be `"~> 1.3 or > 2.0"` is the best solution, because as maintainer if in n-th poison version something will go wrong - you will get an issue and fix it.

So, you can choose the best solution, I'll implement it in this PR.